app/routes/__frontend/resources/developer-resources/open-source/contributor-guide.mdx

```diff
@@ -0,0 +1,178 @@
+
+Unless the documentation says otherwise, you should comment on an issue you want to work on. Ask the maintainer to assign you to it, and only start working on it when they give it to you. It helps to avoid duplication of work and ensures that expectations are aligned. Unless the repository explicitly states that it accepts unsolicited pull requests, you should not take the initiative to make a pull request to a project.
+
+### Communication with Maintainers
+
+Communication is key to working on open source. Most projects have communication conventions, and contributors should always follow these rules. Many projects primarily communicate via comments on issues or pull requests. Some projects have external groups such as Slack, a forum, IRC, or email lists where communication occurs and work is coordinated. If the project uses these, there should be information on the README.md or CONTRIBUTIONS/CONTRIBUTING.md file. Without specific details on where to communicate, you can assume that communication can occur via comments on issues or pull requests.
+
+Regardless of the tools used to communicate, there are certain things to keep in mind to keep communication effective and respectful, especially in an international online space.
+
+- **Always follow the Code of Conduct (COC)** <br />
+  Yes, always, even if you disagree with its contents or think it restricts your freedom of speech. The community agreed on these rules to govern the space. As an entrant to the community, you must follow these rules.
+- **Messages should be kept short and straightforward** <br />
+  It is often best to keep to 'one message, one topic'. However, there may be times when this is not possible. For example, if you will be unavailable for some time and need to respond to several points. If you need to discuss several topics in one message, put each in a separate paragraph.
+- **Keep the conversation respectful** <br />
+  Disagreement and debate are healthy, but it is important to keep the conversation respectful. You could criticize people's ideas but don't criticize the person directly.
+- **Low-context communication in international, cross-cultural spaces** <br />
+  Meanings should be explicit in the words you use. People from different cultures might not understand specific cultural references, cliches, and sayings.
+
+### Self-Care
+
+We recommend you approach open source work with the same professionalism as you would work for an employer. But remember, you are doing this in your spare time, and self-care is essential. It means that you should take care of things such as code style, documentation, and testing, just as you would in your work for an employer, but it is okay if you cannot work at a fast pace. If timescales arise in open source contributions, try to set reasonable expectations from the beginning. Otherwise, take a reasonably relaxed time to work on things. You should not be expected to complete the work to a given timescale for non-trivial issues.
```

Comment:
Suggestios: "We recommend you approach open source work with the same professionalism as you would work for an employer. But remember, you are doing this in your spare time, and self-care is essential. It means that you should take care of things such as code style, documentation, and testing, just as you would in your work for an employer, but it is okay if you cannot work at a fast pace. Both you and the maintainer should set reasonable expectations from the beginning. You should expect to have a reasonably relaxed time to work on things. You should not be expected to complete the work by a strict deadline even for non-trivial issues."

Reply:
Sounds much better! Thanks 😄